pkg/secrets-store/provider_client.go

```diff
@@ -20,63 +20,138 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"net"
+	"os"
+	"regexp"
+	"sync"
 
 	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
 
 	internalerrors "sigs.k8s.io/secrets-store-csi-driver/pkg/errors"
 	"sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1"
 )
 
-// Strongly typed provider name
-type CSIProviderName string
+// ServiceConfig is used when building CSIDriverProvider clients. The configured
+// retry parameters ensures that RPCs will be retried if the underlying
+// connection is not ready.
+//
+// For more details see:
+// https://github.com/grpc/grpc/blob/master/doc/service_config.md
+const ServiceConfig = `
+{
+	"methodConfig": [
+		{
+			"name": [{"service": "v1alpha1.CSIDriverProvider"}],
+			"waitForReady": true,
+			"retryPolicy": {
+				"MaxAttempts": 3,
+				"InitialBackoff": "1s",
+				"MaxBackoff": "10s",
+				"BackoffMultiplier": 1.1,
+				"RetryableStatusCodes": [ "UNAVAILABLE" ]
+			}
+		}
+	]
+}
+`
+
+var (
+	// PluginNameRe is the regular expression used to validate plugin names.
+	PluginNameRe        = regexp.MustCompile(`^[a-zA-Z0-9_-]{0,30}$`)
```

Comment:
We should also add the provider name restrictions in the providers doc in a follow up PR